client/rpc/requestbuilder.go

```diff
@@ -60,7 +62,17 @@ func (r *requestBuilder) Body(body io.Reader) RequestBuilder {
 func (r *requestBuilder) FileBody(body io.Reader) RequestBuilder {
 	pr, _ := files.NewReaderPathFile("/dev/stdin", io.NopCloser(body), nil)
 	d := files.NewMapDirectory(map[string]files.Node{"": pr})
-	r.body = files.NewMultiFileReader(d, false)
+
+	version, err := r.shell.loadRemoteVersion()
+	if err != nil {
+		// Unfortunately, we cannot return an error here. Changing this API is also
+		// not the best since we would otherwise have an inconsistent RequestBuilder.
+		// We save the error and return it when calling [requestBuilder.Send].
+		r.buildError = err
+		return r
+	}
+
+	r.body = files.NewMultiFileReader(d, false, version.LT(ipfs.EncodedAbsolutePathVersion))
```

Comment:
nit: when we have multiple booleans like this, naming them makes it a bit easier to debug in the future
```suggestion
	useEncodedAbsPaths := version.LT(ipfs.EncodedAbsolutePathVersion)
	r.body = files.NewMultiFileReader(d, false, useEncodedAbsPaths)
```